satpy/readers/gerb_l2_hr_h5.py

```diff
@@ -0,0 +1,105 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# Copyright (c) 2023
+#
+# This file is part of satpy.
+#
+# satpy is free software: you can redistribute it and/or modify it under the
+# terms of the GNU General Public License as published by the Free Software
+# Foundation, either version 3 of the License, or (at your option) any later
+# version.
+#
+# satpy is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+# A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along with
+# satpy.  If not, see <http://www.gnu.org/licenses/>.
+
+
+"""GERB L2 HR HDF5 reader.
+
+A reader for the Top of Atmosphere outgoing fluxes from the Geostationary Earth Radiation
+Budget instrument aboard the Meteosat Second Generation satellites.
+"""
+
+
+import logging
+from datetime import timedelta
+
+import dask.array as da
+import h5py
+import xarray as xr
+
+from satpy.readers.hdf5_utils import HDF5FileHandler
+from satpy.resample import get_area_def
+
+LOG = logging.getLogger(__name__)
+
+
+def gerb_get_dataset(hfile, name, ds_info):
+    """
+    Load a GERB dataset in memory from a HDF5 file or HDF5FileHandler.
+
+    The routine takes into account the quantisation factor and fill values.
+    """
+    ds = xr.DataArray(hfile[name][...])
```

Comment:
This is one reason you're getting the dask errors and warnings you're seeing. The result of `hfile[name]` (I think) should be an xarray.DataArray already. And don't do `[...]` or you'll load it into memory which is not what we want anymore now that the reader is trying to be dask friendly. See my other comment for more info...

Reply:
Ok, got it. I am used to h5py and didn't realise that HDF5FileHandler would already take care of that. (+ had a look at the hsaf hdf5 reader that does something similar).